vividus-plugin-web-app/src/main/java/org/vividus/ui/web/expression/DecodeDataUrlExpression.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.ui.web.expression;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.util.Base64;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.Validate;
+import org.vividus.expression.SingleArgExpressionProcessor;
+
+public class DecodeDataUrlExpression extends SingleArgExpressionProcessor<Object>
+{
+    public DecodeDataUrlExpression()
+    {
+        super("decodeDataUrl", DecodeDataUrlExpression::decode);
+    }
+
+    private static Object decode(String dataUrlString)
+    {
+        DataUrl dataUrl = new DataUrl(dataUrlString);
+        Validate.isTrue(dataUrl.data != null, "Supplied argument `%s` is invalid Data URL", dataUrlString);
+        if (dataUrl.base64Encoded)
+        {
+            byte[] decoded = Base64.getDecoder().decode(dataUrl.data.getBytes(UTF_8));
+            return dataUrl.text ? new String(decoded, UTF_8) : decoded;
+        }
+        return dataUrl.data;
+    }
+
+    private static final class DataUrl
+    {
+        private static final Pattern DATA_URL_PATTERN = Pattern.compile("data:([^;]*)?;?([^,]*)?,(.+)");
```

Comment:
SQ violations are reported here, most likely they can be fixed in the following way:
```suggestion
        private static final Pattern DATA_URL_PATTERN = Pattern.compile("data:([^;]*);?([^,]*),(.+)");
```